File: backend/src/main/java/com/huzakerna/cajero/model/StockMovement.java
package com.huzakerna.cajero.model;

import java.math.BigDecimal;
import java.util.UUID;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;

@Entity
@Table(name = "stock_movements")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@SuperBuilder
public class StockMovement extends BaseEntity {

    @Column(name = "ingredient_id")
    private UUID ingredientId;

    @Column(name = "product_id")
    private UUID productId;

    @Column(name = "variant_id")
    private UUID variantId;

    @Column(name = "transaction_id")
    private UUID transactionId;

    @Enumerated(EnumType.STRING)
    @Column(name = "type")
    private StockMovementType type;

    @Column(name = "quantity")
    private BigDecimal quantity;

}

File: mobile/services/types/StockMovement.ts
export interface StockMovement {
	id: string;
	createdAt: string;
	updatedAt: string;
	storeId: string;
	ingredientId: string;
	variantId: string | null;
	productId: string | null;
	transactionId: string | null;
	type: string;
	quantity: number;
	createdBy: string | null;
	updatedBy: string | null;
	createdByName?: string;
	transactionDescription?: string;
	customerName?: string;
}
